src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/FileBasedPrograms/CanonicalMiscFilesProjectLoader.cs

```diff
@@ -29,6 +29,11 @@ internal sealed class CanonicalMiscFilesProjectLoader : LanguageServerProjectLoa
 {
     private readonly Lazy<string> _canonicalDocumentPath;
 
+    /// <summary>
+    /// Avoid showing restore notifications for misc files.
```

Comment:
Consider adding more detail why. Even if it's just that this ended up confusing users